provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/controller/location/LocationController.java

```diff
@@ -1,3 +1,445 @@
+                            .build())
+                    .providerServicesResponse(providerServicesResponse)
+                    .build());
+          }
+        }
+      }
+    }
+    return filteredResults;
+  }
+
+  /** Search by family & given name. */
+  @GetMapping(params = {"address-city"})
+  public Location.Bundle searchByCity(
+      @RequestParam("address-city") String city,
+      @RequestParam(value = "page", defaultValue = "1") @Min(1) int page,
+      @CountParameter @Min(0) int count) {
+    return bundle(
+        Parameters.builder()
+            .add("address-city", city)
+            .add("page", page)
+            .add("_count", count)
+            .build(),
+        page,
+        count);
+  }
+
+  /** Search by Identifier. */
+  @GetMapping(params = {"identifier"})
+  public Location.Bundle searchByIdentifier(
+      @RequestParam("identifier") String identifier,
+      @RequestParam(value = "page", defaultValue = "1") @Min(1) int page,
+      @CountParameter @Min(0) int count) {
+    return bundle(
+        Parameters.builder()
+            .add("identifier", identifier)
+            .add("page", page)
+            .add("_count", count)
+            .build(),
+        page,
+        count);
+  }
+
+  /** Search by name. */
+  @GetMapping(params = {"name"})
+  public Location.Bundle searchByName(
+      @RequestParam("name") String name,
+      @RequestParam(value = "page", defaultValue = "1") @Min(1) int page,
+      @CountParameter @Min(0) int count) {
+    return bundle(
+        Parameters.builder().add("name", name).add("page", page).add("_count", count).build(),
+        page,
+        count);
+  }
+
+  /** Search by family & given name. */
+  @GetMapping(params = {"address-state"})
+  public Location.Bundle searchByState(
+      @RequestParam("address-state") String state,
+      @RequestParam(value = "page", defaultValue = "1") @Min(1) int page,
+      @CountParameter @Min(0) int count) {
+    return bundle(
+        Parameters.builder()
+            .add("address-state", state)
+            .add("page", page)
+            .add("_count", count)
+            .build(),
+        page,
+        count);
+  }
+
+  /** Search by family & given name. */
+  @GetMapping(params = {"address-postalcode"})
+  public Location.Bundle searchByZip(
+      @RequestParam("address-postalcode") String zip,
+      @RequestParam(value = "page", defaultValue = "1") @Min(1) int page,
+      @CountParameter @Min(0) int count) {
+    return bundle(
+        Parameters.builder()
+            .add("address-postalcode", zip)
+            .add("page", page)
+            .add("_count", count)
+            .build(),
+        page,
+        count);
+  }
+
+  private List<LocationWrapper> searchIdentifier(MultiValueMap<String, String> parameters) {
+    LocationWrapper.LocationWrapperBuilder locationWrapper =
+        LocationWrapper.builder()
+            .careSitesResponse(CareSitesResponse.builder().build())
```

Comment:
The idea here is to include empty responses in the `LocationWrapper` to simplify the transformer logic later, so you don't have to check these for nulls?

Lombok has a feature to help with this. In `LocationWrapper`, you can include a builder default value for cases where the field is not set:

` @Builder.Default CareSitesResponse careSitesResponse = CareSitesResponse.builder().build();`